src/screens/augmentedReality/ARShowScreen.js

```diff
@@ -112,20 +116,69 @@ export const ARShowScreen = ({ navigation, route }) => {
   );
 };
 
+/**
+ * index creation function for model and texture
+ *
+ * @param {string} startDate           start date of the model
+ * @param {number} timePeriodInDays    time period in days of the model
+ * @param {array}  scenes              array of models
+ *
+ * @return {object} both parsed values as an object, like { modalIndex: 1, textureIndex: 1 }
+ */
+const multipleSceneIndexGenerator = async ({ startDate, timePeriodInDays, scenes }) =>
+  new Promise((resolve) => {
```

Comment:
why are you using a promise here and async function? i cannot find any async requirements the code mathematics?

Reply:
After running the function, I realised that the application continued without the information I needed from the function and the data was not arriving properly.
So I felt the need to use `async/await`.